### ISSUE TITLE
Exibir diferentes sugestões de título no placeholder ao criar uma publicação

### ISSUE THREAD

**Issue by Rafatcb** [filipedeschamps/tabnews.com.br#1596]
### Descrição

Uma das formas de mostrar ao usuário o que é esperado ao preencher um campo é no `placeholder` do `TextInput`. Conforme discutido em https://github.com/filipedeschamps/tabnews.com.br/pull/1580#discussion_r1437874228, seria interessante ter diferentes sugestões de títulos com base no tipo de publicação que queremos incentivar.

É importante que o texto seja relativamente curto, para que seja inteiramente visível (ou quase) em dispositivos com telas menores. Tenho um celular antigo e com tela pequena que é possível visualizar 47 caracteres, então creio que esse valor possa ser usado como referência, mas isso não é uma exigência. Se o título for maior e der para entender sem ler o final, também é algo positivo.

### Sugestão de implementação

O `placeholder` atual está aqui:

https://github.com/filipedeschamps/tabnews.com.br/blob/769a61b6fd7c65af423b2c54c352a096c560e984/pages/interface/components/Content/index.js#L457

Ele tem 68 caracteres, então se puder ser diminuído um pouco seria algo legal também.

As sugestões podem ficar num array e ser escolhido um elemento de forma aleatória. Se essa abordagem for escolhida, talvez seja necessário fazer uso de um `useMemo` ou `useRef` para evitar trocar o `placeholder` ao rerenderizar o componente `Content`.

Antes da implementação seria bom termos as sugestões de `placeholder`s definidas neste issue.

**Comment by diegosano** [comment]
@Rafatcb gostaria de participar e trabalhar nessa issue

**Comment by Rafatcb** [comment]
@diegosano sem problemas 😄 

Tem alguma sugestão de títulos que podemos exibir para o usuário? Considerando a pergunta [Que tipo de conteúdo eu posso publicar no TabNews?](https://www.tabnews.com.br/faq#conteudo-tabnews) do FAQ, acho que podemos criar uma sugestão para cada "tipo de conteúdo". São 7 tipos listados na resposta do FAQ, e talvez alguns não tenham um bom exemplo de título, mas é um bom guia.

* notícias
* artigos
* tutoriais
* indicações
* curiosidades
* sugestões de software e ferramentas
* perguntas bem formuladas

Também podemos melhorar o placeholder atual ( `e.g. Desafios que tive no meu primeiro ano empreendendo com software`) para ficar menor, como mencionei na criação do issue.

**Comment by diegosano** [comment]
Opa, muito obrigado, acho muito boa a ideia de ter pelo menos 1 título para cada sugestão do FAQ.

- Notícias: "e.g A nova era da computação quântica"
- Artigos: "e.g Desafios para se empreender como desenvolvedor"
- Tutoriais: "e.g Como começar com a linguagem Rust"
- Indicações: "e.g Top 5 livros fundamentais para desenvolvedores"
- Curiosidades: "e.g Fatos curiosos sobre IA que você não sabia"
- Software e ferramentas: "e.g Softwares para melhorar sua produtividade"
- Perguntas bem formuladas: "e.g Como renomear uma branch local no Git?"

 O que acha dessas sugestões?

**Comment by Rafatcb** [comment]
@diegosano gostei das suas sugestões e o comprimento delas está ótimo. Vou tentar complementar com base no que me parece um título de uma publicação que "tem cara de ser boa". Não esqueça que a abreviação no início é `e.g.` (com um ponto no final).

> Notícias: "e.g A nova era da computação quântica"

Me pareceu muito click bait. Acho que notícia será o tipo mais difícil de criar uma sugestão de título sem ficar algo muito datado e ao mesmo tempo fazer sentido.

> Artigos: "e.g Desafios para se empreender como desenvolvedor"

Tive a impressão de que aqui misturou dois possíveis títulos, que são assuntos diferentes: _"Desafios ao empreender como desenvolvedor"_ e _"Problemas para solucionar como desenvolvedor"_ (ou algo parecido). A primeira opção cita diretamente dois grupos que o TabNews gosta de incentivar (por meio dos Pitchs), então eu escolheria ela. Acho que o título está muito abrangente, mas agora não consegui pensar em algo melhor e que continue curto.

> Tutoriais: "e.g Como começar com a linguagem Rust"

Eu gostei, mas creio que se formos mais específicos, podemos incentivar publicações mais aprofundadas. Pensei em algo como `e.g. Como funciona o conceito de ownership em Rust`. Faz sentido?

> Indicações: "e.g Top 5 livros fundamentais para desenvolvedores"

:+1: Acho que está bom. Já tivemos publicações sobre livros, e geralmente outras pessoas agregam nos comentários com outras sugestões ou experiências que tiveram com os livros citados.

> Curiosidades: "e.g Fatos curiosos sobre IA que você não sabia"

Acho que isso ficou com cara de "Fatos Desconhecidos" :laughing:. Podemos tentar um título mais específico.

> Software e ferramentas: "e.g Softwares para melhorar sua produtividade"

[Software não tem plural](https://www.dictionary.com/e/is-software-a-mass-noun/). Fora isso, achei o título bom. Pode ser "e.g. Aplicativos para melhorar sua produtividade".

> Perguntas bem formuladas: "e.g Como renomear uma branch local no Git?"

Essa é uma pergunta que poderia ser facilmente feita para o Google, mas não está mal feita. Talvez alguém consiga sugerir algo melhor.

---

Um pensamento um pouco off-topic que passou pela minha cabeça, quando termos tags (https://github.com/filipedeschamps/tabnews.com.br/issues/742) poderemos usar um placeholder na tag que faça sentido com o título que está aparecendo, então isso ficará "ainda mais claro", pois terá mais contexto.

**Comment by diegosano** [comment]
@Rafatcb concordo com o que você complementou, acredito que por enquanto temos:

- Notícias: "e.g. A nova era da computação quântica" _*procurando uma sugestão melhor_
- Artigos: "e.g. Desafios ao empreender como desenvolvedor"
- Tutoriais: "e.g. Como funciona o conceito de ownership em Rust"
- Indicações: "e.g. Top 5 livros fundamentais para desenvolvedores"
- Curiosidades: "e.g. Fatos curiosos sobre IA que você não sabia" _*procurando algo menos "Fatos desconhecidos"_ 😄  _e mais específico_
- Software e ferramentas: "e.g. Aplicativos para melhorar sua produtividade"
- Perguntas bem formuladas: "e.g. Como renomear uma branch local no Git?" _*procurando um título de pergunta mais específica, que não se encontre no Gooogle, por exemplo_

Acho que estamos em um bom caminho, vou continuar pensando e quem sabe mais alguém da comunidade possa ajudar com essa parte criativa! 

Vou começar a implementar o código, acredito que ao longo do desenvolvimento essas sugestões apareçam e podemos ir modificando esse array de exemplos, o que acha?

**Comment by Rafatcb** [comment]
@diegosano acredito que temos uma base boa. Pode fazer, talvez o PR chame mais atenção para outras pessoas avaliarem :+1:
